src/ast/passes/codegen_llvm.cpp

```diff
@@ -2428,7 +2432,25 @@ void CodegenLLVM::maybeAllocVariable(const std::string &var_ident,
   }
 
   AllocaInst *val = b_.CreateAllocaBPFInit(alloca_type, var_ident);
-  variables_[var_ident] = VariableLLVM{ val, val->getAllocatedType() };
+  variables_[scope_stack_.back()][var_ident] = VariableLLVM{
+    val, val->getAllocatedType()
+  };
+}
+
+VariableLLVM &CodegenLLVM::getVariable(const std::string &var_ident)
+{
+  for (auto scope : scope_stack_) {
```

Comment:
Correct me if I'm wrong but don't we need to iterate through scope_stack_ in reverse order e.g. from deepest scope -> top-level scope.

For example:
```
{
  $ x= 1
  if (1) {
    let $x;
    $x = 2;
  }
}
```

In that case, I think the current code would return the variable `$x` in the top-level scope in all cases.

Reply:
It's a good question but we don't need to do this because bpftrace doesn't support variable shadowing. So that script above should throw an error.